src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -21,39 +45,311 @@ const ActorInit En_Ani_InitVars = {
+    ICHAIN_F32(unk_F4, 850, ICHAIN_STOP),
+};
+
+static Vec3f EnAniVec = {
+    800.0f,
+    500.0f,
+    0.0f,
+};
+
+UNK_PTR D_809B0F80[] = {
+    0x06000408,
+    0x06001518,
+    0x06001D18,
+    0x00000000,
+};
+
+void EnAni_SetupAction(EnAni* this, ActorFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnAni_Init(EnAni* this, GlobalContext* globalCtx) {
+    AnimationHeader* anim;
+    u32 pad;
+
+    anim = &D_060076EC;
+    Actor_ProcessInitChain(&this->actor, initChain);
+    ActorShape_Init(&this->actor.shape, -2800.0f, ActorShadow_DrawFunc_Circle, 36.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060000F0, anim, &this->unk_1DC, &this->unk_23C, 0x10);
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, anim);
+    ActorCollider_AllocCylinder(globalCtx, &this->collider);
+    ActorCollider_InitCylinder(globalCtx, &this->collider, &this->actor, &cylinderInitData);
+    this->actor.sub_98.mass = 0xFF;
+    if (LINK_IS_CHILD) {
+        EnAni_SetupAction(this, func_809B064C);
+    } else {
+        EnAni_SetupAction(this, func_809B07F8);
+    }
+    this->unk_2AA = 0;
+    this->unk_2A8 = 0;
+    this->actor.minVelocityY = -1.0f;
+    this->actor.velocity.y = -1.0f;
+}
+
+void EnAni_Destroy(EnAni* this, GlobalContext* globalCtx) {
+    ColliderCylinderMain* collider;
+    collider = &this->collider;
+    ActorCollider_FreeCylinder(globalCtx, &this->collider);
+}
+
+s32 EnAni_SetText(EnAni* this, GlobalContext* globalCtx, u16 textId) {
+    this->actor.textId = textId;
+    this->unk_2A8 |= 1;
+    func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    return 0;
+}
+
+void func_809B04F0(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx) != 0) {
+        EnAni_SetupAction(this, func_809B064C);
+    }
+}
+
+void func_809B0524(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx) != 0) {
+        EnAni_SetupAction(this, func_809B07F8);
+    }
+}
+
+void func_809B0558(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F410(&this->actor, globalCtx) != 0) {
+        this->actor.attachedA = 0;
+        if (LINK_IS_CHILD) {
+            EnAni_SetupAction(this, func_809B04F0);
+        } else {
+            EnAni_SetupAction(this, func_809B0524);
+        }
+        gSaveContext.item_get_inf[1] = gSaveContext.item_get_inf[1] | 0x20;
+        return;
+    }
+    func_8002F434(&this->actor, globalCtx, GI_HEART_PIECE, 10000.0f, 200.0f);
+}
+
+void func_809B05F0(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F334(this, globalCtx) != 0) {
+        EnAni_SetupAction(this, func_809B0558);
+    }
+    func_8002F434(this, globalCtx, GI_HEART_PIECE, 10000.0f, 200.0f);
+}
+
+void func_809B064C(EnAni* this, GlobalContext* globalCtx) {
+    u16 textId;
+    s16 rotDiff;
+    u16 textId2;
+
+    textId2 = Text_GetFaceReaction(globalCtx, 0xA);
+    textId = textId2 & 0xffff;
+
+    if (!textId) {}
```

Comment:
can this just be if (1) {} ? sometimes the permuter uses a variable for this when its not really needed

Reply:
maybe if (0) makes more sense with the not there, idk. if neither work though then its whatever

Reply:
no, if(1){} doesnt work. using the variable shifts regalloc. there may be a "better" way to do it, but i didnt find it

Reply:
alright no worries then